File: backend/app/schemas/meeting.py
"""Meeting Pydantic schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import MeetingStatus


class MeetingCreateRequest(BaseModel):
    file_id: uuid.UUID


class MeetingResponse(BaseModel):
    id: uuid.UUID
    chat_id: uuid.UUID
    file_id: uuid.UUID
    title: str | None
    status: MeetingStatus
    duration_seconds: float | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/meeting.ts
export type MeetingStatus = 'pending' | 'processing' | 'ready' | 'failed' | 'cancelled';

export interface Meeting {
  id: string;
  chat_id: string;
  file_id: string;
  title: string | null;
  status: MeetingStatus;
  duration_seconds: number | null;
  created_at: string;
}
